pkg/tasks/queue/pgqueue.go

```diff
@@ -434,12 +460,16 @@ func (p *PgQueue) taskDependents(ctx context.Context, conn Connection, id uuid.U
 }
 
 func (p *PgQueue) Status(taskId uuid.UUID) (*models.TaskInfo, error) {
-	var conn Connection
 	var err error
-	conn = p.Conn
 
 	// Use double pointers for timestamps because they might be NULL, which would result in *time.Time == nil
 	var info models.TaskInfo
+	conn, err := p.Pool.Acquire(context.Background())
+
+	defer conn.Release()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I think it would be better to check for an error before deferring release. I'm not sure what happens if there's an error and you release the connection.

Reply:
agreed! good point